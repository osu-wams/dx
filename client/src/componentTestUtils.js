import React from 'react';
import { render as rtlRender } from 'react-testing-library';
import { createHistory, createMemorySource, LocationProvider } from '@reach/router';
import { ThemeProvider } from 'styled-components';
import theme from './theme';

function render(ui, ...rest) {
  return rtlRender(<ThemeProvider theme={theme}>{ui}</ThemeProvider>, ...rest);
}

// this is a handy function that I would utilize for any component
// that relies on the router being in context
function renderWithRouter(
  ui,
  { route = '/', history = createHistory(createMemorySource(route)) } = {}
) {
  return {
    ...rtlRender(<LocationProvider history={history}>{ui}</LocationProvider>),
    // adding `history` to the returned utilities to allow us to reference it in our tests
    history
  };
}

export * from 'react-testing-library';
// override the built-in render with our own
export { render, renderWithRouter };
