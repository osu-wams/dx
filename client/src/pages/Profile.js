import React, { Component } from 'react';
import PageTitle from '../ui/PageTitle';

export default class Profile extends Component {
  constructor(props) {
    super(props);
    this.state = { api: 'Api' };
  }

  render() {
    const { api } = this.state;
    return (
      <div data-testid="profile-page">
        <PageTitle title="My Profile" />
        <p>Profile {api} </p>
      </div>
    );
  }
}
