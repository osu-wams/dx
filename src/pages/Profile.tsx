import React from 'react';
import PageTitle from '../ui/PageTitle';
import OSUProfile from '../features/profile/OSUProfile';
import { MainGridWrapper, MainGrid } from '../ui/PageGrid';

const Profile = () => (
  <div data-testid="profile-page">
    <MainGridWrapper>
      <PageTitle title="My Profile" />
      <MainGrid>
        <OSUProfile />
      </MainGrid>
    </MainGridWrapper>
  </div>
);

export default Profile;
