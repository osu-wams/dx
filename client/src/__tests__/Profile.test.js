import React from 'react';
import { render } from '../componentTestUtils';
import Profile from '../components/pages/Profile';

test('renders', () => {
  const { container } = render(<Profile />);
  expect(container.firstChild).toMatchSnapshot();
});
