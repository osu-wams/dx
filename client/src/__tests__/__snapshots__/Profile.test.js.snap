// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<div
  data-testid="profile-page"
>
  <div
    style="border: 0px; height: 1px; width: 1px; margin: -1px; padding: 0px; overflow: hidden; position: absolute;"
  >
    <h2>
      My Profile
    </h2>
  </div>
  <p>
    Profile 
    Api
     
  </p>
</div>
`;
